flutter/lib/src/view_hierarchy/sentry_tree_walker.dart

```diff
@@ -0,0 +1,75 @@
+import 'dart:typed_data';
+
+import 'package:flutter/widgets.dart';
+
+import '../../sentry_flutter.dart';
+import 'sentry_view_hierarchy.dart';
+import '../widget_utils.dart';
+
+class _TreeWalker {
+  final Element rootElement;
+
+  _TreeWalker(this.rootElement);
+
+  ValueChanged<Element> _visitor(SentryViewHierarchyElement parentNode) {
+    return (Element element) {
+      final node = _toSentryViewHierarchyElement(element);
+      parentNode.children.add(node);
+      element.visitChildElements(_visitor(node));
+    };
+  }
+
+  SentryViewHierarchy? toSentryViewHierarchy() {
+    final rootNode = _toSentryViewHierarchyElement(rootElement);
+    rootElement.visitChildElements(_visitor(rootNode));
```

Comment:
@ueman so the result of this for simple screens is quite deep in-depth, and it looks way more deeper in depth than the view hierarchy of the dev tooling, have you noticed that, or did I mess up something?

Reply:
Unfortunately yes. See https://github.com/getsentry/team-mobile/issues/64#issuecomment-1284547394 for some ideas on how to tackle that.

Reply:
I wonder if the `_` check for private classes also works in obfuscated code

Reply:
I will check, also found this https://docs.flutter.dev/development/tools/devtools/inspector#track-widget-creation

Reply:
There are also a few methods that print the tree, but only on debug mode https://docs.flutter.dev/testing/code-debugging#debug-flags-performance

Reply:
@ueman That does not work when obfuscated is enabled because everything is renamed at compile time.
Still worth keeping this quick workaround; it improves at least when it's disabled.